lab03/list_map.s

```diff
@@ -0,0 +1,126 @@
+.globl map
+
+.text
+main:
+    jal ra, create_default_list
+    add s0, a0, x0 # a0 = s0 is head of node list
+
+    #print the list
+    add a0, s0, x0
+    jal ra, print_list
+
+    # print a newline
+    jal ra, print_newline
+
+    # load your args
+    add a0, s0, x0 # load the address of the first node into a0
+
+    # load the address of the function in question into a1 (check out la on the green sheet)
+    ### YOUR CODE HERE ###
+
+    # issue the call to map
+    jal ra, map
+
+    # print the list
+    add a0, s0, x0
+    jal ra, print_list
+
+    # print another newline
+    jal ra, print_newline
+
```

Comment:
Suggestion: Add another call to map here with a different function input (say, add 1). The most common error was students using la a1 square in the map function anyway (and they likely won't read the comment), and this catches that for them.

Reply:
Good idea, I added that as a sanity check.